templates/.github/workflows/js-test-and-release.yml

```
@@ -0,0 +1,125 @@
+name: test & maybe release
+on:
+  push:
+    branches:
+      - $default-branch # with #262 - ${{{ github.default_branch }}}
+  pull_request:
+    branches:
+      - $default-branch # with #262 - ${{{ github.default_branch }}}
+
+jobs:
+
+  check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: lts/*
+    - uses: ipfs/aegir/actions/cache-node-modules@master
+    - run: npm run --if-present lint
+    - run: npm run --if-present dep-check
+
+  test-node:
+    needs: check
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: [16]
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - run: npm run --if-present test:node
+      - uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192 # v1
```

v1 will stop functioning in a couple of weeks - https://github.com/codecov/codecov-action#%EF%B8%8F--deprecration-of-v1

```suggestion
      - uses: codecov/codecov-action@v2
```

---

We should include this by hash to prevent supply-chain attacks. Can we use the same version that we use for the Go workflow?

---

I upgraded the action to `@v2.1.0` - same as we use in go workflows.

---

If we're worried about malicious patch releases etc, shouldn't all the other actions have full version specifiers?  `actions/checkout`, `actions/setup-node` etc just have `v2`.

---

Also, I mean, we switched `actions/checkout@2.4.0` for `actions/checkout@v2` - 
https://github.com/protocol/.github/pull/274/files/5cec3f5077e1d6ce52c17c73c85460793085feaf#r781040115

Are we only concerned about supply-chain attacks for some actions and not others?

---

> Are we only concerned about supply-chain attacks for some actions and not others?

Yes, here's a list of who we decided to trust https://github.com/protocol/.github/blob/master/.github/workflows/check-3rd-party.sh#L10-L13 (tldr it's us and GitHub).